File: zjuctf2022/babyre/solve.py
from operator import xor
from pysm4 import *
from pwn import *
key = 'xpLBYaJVKyjX^cwd'
xorresult = b'\x36\x6D\xED\x6B\xFE\x88\x3B\xC9\x01\xA5\x37\xA5\x03\xC2\x3C\x02'
v6 = b'\x64\x08\x9B\x58\xCA\xBD\x5E\x96\x30\xD6\x68\xC0\x62\xF7\x45\x23'
buffer = bytes(a ^ b for a, b in zip(v6, xorresult))
print(buffer)

File: zjuctf2022/babyre/solve.c

#include <stdio.h>
#include "sm4.h"
char *key = "xpLBYaJVKyjX^cwd";
char cmp[16] = {0x36, 0x6D, 0xED, 0x6B, 0xFE, 0x88, 0x3B, 0xC9, 0x01, 0xA5, 0x37, 0xA5, 0x03, 0xC2, 0x3C, 0x02};
char 
int main()
{
    char output[16];
    sm4_context ctx;
    sm4_setkey_enc(&ctx, key);
    sm4_crypt_ecb(&ctx, SM4_ENCRYPT, 16, key, output);
    for(int i=0;i<16;i++){
        output[i]^=cmp[i];
    }
    printf("%s", output);
}